src/circulating_supply/circulating_supply_provider.rs

```diff
@@ -0,0 +1,150 @@
+use bigdecimal::{BigDecimal, ToPrimitive};
+use diesel::PgConnection;
+
+use crate::circulating_supply::lockup::{
+    get_account_state, get_code_version, is_bug_inside, TRANSFERS_ENABLED,
+};
+use crate::circulating_supply::user_balance::get_user_balance;
+use crate::db_adapters::accounts::collect_lockups_for_block;
+use crate::db_adapters::blocks::closest_block_for;
+use crate::db_adapters::circulating_supply::{
+    add_circulating_supply, get_precomputed_circulating_supply,
+};
+use crate::models;
+use crate::models::circulating_supply::CirculatingSupply;
+use crate::models::Block;
+use actix::Addr;
+use actix_web::rt::time;
+use std::ops::Div;
+use std::str::FromStr;
+use std::time::SystemTime;
+use tracing::info;
+
+const DAY: std::time::Duration = std::time::Duration::from_secs(86400);
+
+pub(crate) async fn compute_circulating_supply(view_client: Addr<near_client::ViewClientActor>) {
+    let mut current_day = TRANSFERS_ENABLED;
+    while current_day < get_now_timestamp() {
+        compute_circulating_supply_for(&view_client, current_day).await;
+        current_day += DAY.as_nanos() as u64;
```

Comment:
How long does it take to recompute all the history? I feel that might need to be a separate mode triggered explicitly since we reboot indexers somewhat often and this might be wasteful.

Reply:
Todo: put a check here if it was already computed
and delete this check from the algorithm itself